pkg/controller/idler/idler_controller.go

```diff
@@ -53,7 +53,7 @@ func add(mgr manager.Manager, r *ReconcileIdler) error {
 
 	// Watch for changes to primary resource Idler
 	if err := c.Watch(
-		source.NewKindWithCache(&toolchainv1alpha1.Idler{}, r.cache),
+		&source.Kind{Type: &toolchainv1alpha1.Idler{}},
```

Comment:
hmm, I thought this was needed, but maybe not anymore after this refactoring? 🤔

Reply:
Originally, you used for watching the "all namespaces" client that is created on our own, not that one that was delivered via the Manager. In my code, I use the "standard" namespace-scoped client for watching the Idlers and the "all namespaces" one only for manipulating the payloads.

Reply:
ah ok, I did not notice that. Makes sense now!